File: Armstrong-Numbers/Armstrong.java
/*What is a Armstrong Number?*/
/*Well in case of an Armstrong number of 3 digits, the sum of cubes of each digits is equal to the number itself.*/

import java.util.Scanner;

public class Armstrong {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int inp = input.nextInt();
        boolean ans = IsArmstrong(inp);
        System.out.println(ans);


        for (int i = 0 ; i < 1000 ; i++) {
            if (IsArmstrong(i)) {
                System.out.println(i + " ");
            }


        }
    }
    static boolean IsArmstrong(int inp) {
        int num = inp;
        int sum = 0;

        while (inp > 0) {
            int remain = inp % 10;
            inp = inp / 10;
            sum = sum + remain * remain * remain;
            }
        return sum == num;
    }
}

File: Armstrong-Numbers/Armstrong.kt
import java.util.*

object Armstrong {
    @JvmStatic
    fun main(args: Array<String>) {
        val input = Scanner(System.`in`)
        val inp = input.nextInt()
        val ans = IsArmstrong(inp)
        println(ans)
        for (i in 0..999) {
            if (IsArmstrong(i)) {
                println("$i ")
            }
        }
    }

    fun IsArmstrong(inp: Int): Boolean {
        var inp = inp
        val num = inp
        var sum = 0
        while (inp > 0) {
            val remain = inp % 10
            inp = inp / 10
            sum = sum + remain * remain * remain
        }
        return sum == num
    }
}